flow/project.py

```diff
@@ -2021,8 +2245,31 @@ def print_status(self, jobs=None, overview=True, overview_max_lines=None,
             file = sys.stdout
         if err is None:
             err = sys.stderr
-        if jobs is None:
-            jobs = self     # all jobs
+
+        if jobs is not None:
+            aggregates = set()  # Set in order to prevent duplicate entries
+            for job in jobs:
+                if isinstance(job, signac.contrib.job.Job):  # User can still pass signac jobs
+                    aggregates.add((job,))
+                else:
+                    try:
+                        aggregates.add(tuple(job))  # An aggregate provided by the user
+                    except Exception:
+                        raise ValueError('Invalid aggregate provided by the user. If you want to '
+                                         'provide an aggregate, please pass a tuple of jobs '
+                                         'instead.')
+        else:
+            aggregates = None
+
```

Comment:
This logic exists in multiple places. Placing this in a function would be helpful.